sdcm/cluster.py

```diff
@@ -3055,11 +3056,12 @@ def __init__(self, cluster_uuid=None, cluster_prefix='cluster', node_prefix='nod
         # NOTE: following is needed in case of K8S where we init multiple DB clusters first
         #       and only then we add nodes to it calling code in parallel.
         if add_nodes:
+            add_nodes_func = self.add_db_nodes if 'db' in self.node_type else self.add_nodes
```

Comment:
this code would execute only for first node added to the clusters (and also won't execute on k8s, according to the comment here)

any nodes created by nemesis code, won't be passing via this part...

Reply:
i.e. cluster.add_nodes would still be called from lots of places without passing via the newly introduced `add_db_nodes`

Reply:
I would recommend creating `add_db_nodes` as a decorator and warp `add_nodes` on cluster __init__ code.

Reply:
@fruch , i think it is a wrong assumption about the nemesis because you can simply see in the above logs that it did collect 2 added nodes (of a total 8 where 6 initiated in setup).

Reply:
@fruch i'm not familiar with k8s, why do you think it doesn't work in this case? perhaps i can run a test for it to check as well.

Reply:
@fruch ' i'm not aware of any dynamic node creation without using `add_db_nodes`, can you please point out an example? 
i guess of course there could be some exceptions as for siren or k8s, but might not be crittical.

Reply:
There are other places in the code add_nodes is being called, see longevity for the 5000 keyspace, the scaling tests, k8s functional tests and so on.

I don't want a change like this that forces us to switch API usage across the board like that, it would just adds unneeded confusion.

As for decorating, I've meant applying the decorator at run time in BaseCluster or BaseScyllaCluster constructor.

One other option is to add this into one of node_setup steps, only for the relevent cluster type.